xbmc/cores/VideoPlayer/DVDMessageQueue.cpp

```diff
@@ -150,9 +169,11 @@ MsgQueueReturnCode CDVDMessageQueue::Get(CDVDMsg** pMsg, unsigned int iTimeoutIn
 
   while (!m_bAbortRequest)
   {
-    if (!m_list.empty() && m_list.back().priority >= priority)
+    std::list<DVDMessageListItem> &msgs = (priority > 0 || !m_prioMessages.empty()) ? m_prioMessages : m_messages;
```

Comment:
prio messages are always considered first, even it player requests a message with prio 0